### PR TITLE
Created spinner component

### LINKED ISSUE #269: Spinner component
Blocks #270 

### Is your feature request related to a problem? Please describe ⚡️
We're missing ways to indicate a loading state when saving a form or validating a textfield. Right now a text replacement like "...Almost done" along with a disabled state is used, but this is not ideal. The background switch is too heavy and with long during requests it looks like the application froze.

### Describe the solution you'd like ✨
To indicate loading states with continuous feedback we want to use an animated spinner. Basically we want this: https://evergreen.segment.com/components/spinner. It uses a svg with a circle component which has an css animation on the stroke-dash-offset for the elastic effect. It should also have a `color` prop so it can be used in different contexts.

### Describe alternatives you've considered 🎡
A progressbar would also be nice, but not suitable for an indeterminate loading state

### PR DESCRIPTION
**Pull request by nicokoelewijn**
### This PR:

resolves #269

proposed release: `patch`

**Checklist** 🛡
- [x] I have exported my addition from `src/index.ts` (check if not applicable)
- [x] Appropriate tests have been added for my functionality (check if not applicable)
- [x] A designer has seen and approved my changes (check if not applicable)
- [x] I have tested my addition in all supported browsers and for responsiveness (Chrome, Firefox, Safari, Edge, IE11 and mobile browsers)

**Adds** ✨
- Spinner component

### CODE REVIEW

File: src/components/Spinner/index.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+const Spinner: FunctionComponent<SpinnerPropsType> = (props): JSX.Element => (
+    <StyledSpinner aria-busy="true" aria-live="polite">
+        <SpinnerSvg
+            // @ts-ignore
+            x="0px"
+            y="0px"
+            viewBox="0 0 150 150"
+            fill="transparent"
+            color={props.color}
```
[the remark this answers is not in the document]
Comment (hborghols) [reply src/components/Spinner/index.tsx]: Good point, I would verify this with design

File: src/components/Spinner/index.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React, { FunctionComponent } from 'react';
+import StyledSpinner, { SpinnerSvg, SpinnerCircle, SpinnerPropsType } from './style';
+
+const Spinner: FunctionComponent<SpinnerPropsType> = (props): JSX.Element => (
+    <StyledSpinner aria-busy="true" aria-live="polite">
+        <SpinnerSvg
+            // @ts-ignore
+            x="0px"
```
Comment (hborghols) [inline src/components/Spinner/index.tsx]: Why do you need this (and the `y` below)? It seems to work without them for me.

File: src/components/Spinner/index.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+            y="0px"
+            viewBox="0 0 150 150"
+            fill="transparent"
```
Comment (hborghols) [inline src/components/Spinner/index.tsx]: You also set this in `style.tsx`, I would remove one of the two

File: src/components/Spinner/index.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import React, { FunctionComponent } from 'react';
+import StyledSpinner, { SpinnerSvg, SpinnerCircle, SpinnerPropsType } from './style';
+
+const Spinner: FunctionComponent<SpinnerPropsType> = (props): JSX.Element => (
```
Comment (woutsluiter) [inline src/components/Spinner/index.tsx]: I think you forgot a PropsType

File: src/components/Spinner/story.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import { storiesOf } from '@storybook/react';
+import Box from '../Box';
+import React from 'react';
+import Spinner from '.';
+
+storiesOf('Spinner', module).add('Default', () => {
+    const size = text('size', '100px');
+
+    return (
+        <Box width={size} height={size}>
```
[the remark this answers is not in the document]
Comment (nicokoelewijn) [reply src/components/Spinner/story.tsx]: I made it the same way the skeleton rectangle is made, with only the core functionality and no unnecessary positioning props. But i'll add an indication that it's part of the story.

File: src/components/Spinner/index.tsx (hunk 2 above)
Comment (nicokoelewijn) [reply src/components/Spinner/index.tsx]: This is based on the https://evergreen.segment.com/components/spinner/ component, but i can't find any added value either, so i'll remove it.

File: src/components/Spinner/index.tsx (hunk 4 above)
Comment (nicokoelewijn) [reply src/components/Spinner/index.tsx]: No i wrote it in the style, but i'll move it to the index.

File: src/components/Spinner/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (nicokoelewijn) [reply src/components/Spinner/index.tsx]: Luuk approves. It should match the font color. So if the component remains as is, the color prop can be passed on to it

File: src/components/Spinner/index.tsx (hunk 4 above)
Comment (hborghols) [reply src/components/Spinner/index.tsx]: Can you rename it to `PropsType` to make it consistent with all other components?

**Pull request merged**